spring-integration-core/src/main/java/org/springframework/integration/channel/AbstractMessageChannel.java

```diff
@@ -173,16 +175,28 @@ public void setDatatypes(Class<?>... datatypes) {
 	public void setInterceptors(List<ChannelInterceptor> interceptors) {
 		Collections.sort(interceptors, this.orderComparator);
 		this.interceptors.set(interceptors);
+		populateBeanFactoryToInterceptors(interceptors.toArray(new ChannelInterceptor[interceptors.size()]));
+	}
+
+	protected void populateBeanFactoryToInterceptors(ChannelInterceptor... interceptors) {
```

Comment:
M-m-m. Sorry, my bad.
See the test-case:
```
queueChannel.addInterceptor(new WireTap("wireTapFromOutput"));
```
And that is the case for the DSL `.wireTap()` variant, too.

Since we don't declare `WireTap` as a bean, the `beanFactory` won't be populated.
I found only this hook to do that manually from the parent component.

Another variant to do the same is like we are going to do in the DSL - add method like `AbstractMessageChannel.addWireTap(String channelName)` :smile: 

(And yes: I've noticed this your comment only from GH logs, not from the email. So, let me know that you see this mine.)